### DISCUSSION TITLE
Attachments image to REST API and getting Response Data

### DISCUSSION THREAD

**Discussion by yeolwoosung** [Budibase/budibase#6699] [Help] [answered]
Dear everyone.

I have a server that can send images to cloud storage and receive stored CDN based url.

The problem I want to solve is:

1. Attach an image in budibase and POST this image to the image server specified above through Define Action.
2. I want to dynamically put the resulting url value into a Text Field.

![image](https://trend.gollala.com/api/cdn/public/view?url=/tmp/20220714/PgvyrZMHQ1N9UbbUwvJk.png)

I'm not sure how I can pass values in this process.
Please let me know if you have a good solution.

Thank you.

**Comment by melohagan** [comment, marked as answer]
Hi @SungYeolWoo 

You can try this:

1. Select the attachment field
2. Click **Define actions** under *On Change*
![Screenshot 2022-07-18 at 08 07 20](https://user-images.githubusercontent.com/101575380/179460248-2024d54c-9c3c-4448-a0ba-cd1271481f1b.png)

3. Choose **Update Field Value** and fill out as follows:
![Screenshot 2022-07-18 at 08 07 57](https://user-images.githubusercontent.com/101575380/179460351-8a077496-0f74-47f9-8dc2-dcf5f65d11ba.png)

In this case, `Text` is the name of the text field.
When a file is dragged into the attachment upload field, it will set the text field to the URL of the first image attachment.

If you wanted to set it to the last uploaded attachment, then you could use the following JavaScript binding:
`return $("Field Value")[$("Field Value").length-1].url;`

**Reply by yeolwoosung** [reply]
Hi, @melohagan Thank you for answer. It helped a lot :)

**Reply by melohagan** [reply]
You're welcome 😄
